resources/lib/tubecast/youtube/app.py

```diff
@@ -430,30 +451,53 @@ def __monitor_volume(self):
 class YoutubeListener(threading.Thread):
 
     def __init__(self, app, ssdp=True):
-        threading.Thread.__init__(self)
-        self.app = app
+        super(YoutubeListener, self).__init__(name="YoutubeListener")
+        self.app = app  # type: YoutubeCastV1
         self.stop = False
         self.ssdp = ssdp
+        self.r = None  # type: Optional[requests.Response]
```

Comment:
Why does this need to be a class var. Can't r be assigned only on the scope of the `with` statement block?

Reply:
Nevermind, you use it to access the underlying socket object.

Reply:
Additionally, this has always been a classvar (the context manager binds to self.r). I just decided to initialise it to remove the possibility of running into an `AttributeError`.